awx/api/serializers.py

```diff
@@ -2825,6 +2826,25 @@ def get_summary_fields(self, obj):
         d['recent_jobs'] = self._recent_jobs(obj)
         return d
 
+    def validate(self, attrs):
+        webhook_service = attrs.get('webhook_service', getattr(self.instance, 'webhook_service', None))
+        webhook_credential = attrs.get('webhook_credential', getattr(self.instance, 'webhook_credential', None))
+
+        if webhook_credential:
+            if not webhook_credential.credential_type.namespace.endswith('_token'):
```

Comment:
I think @AlanCoding may have commented on this already and suggested something like this, but I think that `kind == 'token'` *is* more appropriate here.

https://github.com/ansible/awx/pull/4436/files#diff-8923e305beef7163bf0fcb6d87998376R991

`CredentialType`s *do* have a  `kind`; `Credential.kind` is the type of thing we've been trying to shy away from.  @AlanCoding thoughts?

Reply:
Yeah, that would be undoing exactly what Alan suggested.

Reply:
Alright, @AlanCoding, defend your thesis.

Reply:
Sorry, I believe I did misunderstand the function at the time. credential type kind is like `scm`, `vault`, `ssh`, `insights`, and the kind here is `token`...

I need to consult on words to use for things. This is a new category of credential, that does make sense to me. Its function is to update the status of CI checks (these do not feel like good words). Maybe `ct.kind == 'webhook'` would be readable.